akka-actor/src/main/scala/akka/actor/Address.scala

```diff
@@ -65,6 +68,13 @@ final case class Address private (protocol: String, system: String, host: Option
    * `system@host:port`
    */
   def hostPort: String = toString.substring(protocol.length + 3)
+
+  /** INTERNAL API
+   * Check if the address is not created through `AddressFromURIString`, if there
+   * are any unusual characters in the host string.
+   */
+  @InternalApi
+  private[akka] def hasInvalidHostCharacters: Boolean = host.exists(_.contains("_"))
```

Comment:
Could expand of course but not all cases would need all char checks.
Could in future allow a user to provide a regex but also sounds over-engineery.
Best option is have a base here, and in a given module/function add the additional applicable checks.